railties/lib/rails/application.rb

```diff
@@ -176,10 +176,10 @@ def key_generator
     #     # => 'my sensible data'
     #
     # See the ActiveSupport::MessageVerifier documentation for more information.
-    def message_verifier(verifier_name)
+    def message_verifier(verifier_name, urlsafe: false)
       @message_verifiers[verifier_name] ||= begin
         secret = key_generator.generate_key(verifier_name.to_s)
-        ActiveSupport::MessageVerifier.new(secret)
+        ActiveSupport::MessageVerifier.new(secret, urlsafe: urlsafe)
```

Comment:
Thank you for the pull request! :smiley:  However, this would cause issues with memoziation.  For a given `verifier_name`, the value of `urlsafe` will always be the first specified value.

For example, if some code calls `Rails.application.message_verifier("my_verifier")`, and then some other code calls `Rails.application.message_verifier("my_verifier", urlsafe: true)`, the 2nd call will return a verifier with `urlsafe: false`.

For configuring memoized verifiers (and encryptors), I would like #44179 or something similar.